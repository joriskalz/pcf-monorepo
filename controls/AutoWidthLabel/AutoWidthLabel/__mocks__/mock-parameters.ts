/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
