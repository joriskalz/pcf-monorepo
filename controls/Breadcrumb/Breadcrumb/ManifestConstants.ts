export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Clickable = 'ItemClickable',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
    SelectedKey = 'SelectedKey',
}
