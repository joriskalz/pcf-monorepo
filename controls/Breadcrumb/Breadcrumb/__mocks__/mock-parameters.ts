/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
