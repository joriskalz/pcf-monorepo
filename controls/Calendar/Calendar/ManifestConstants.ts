export const enum InputEvents {
    SetFocus = 'SetFocus',
}
