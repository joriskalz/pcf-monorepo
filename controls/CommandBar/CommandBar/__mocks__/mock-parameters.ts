/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
