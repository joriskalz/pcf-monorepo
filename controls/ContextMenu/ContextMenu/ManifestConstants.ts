export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Enabled = 'ItemEnabled',
    IconName = 'ItemIconName',
    IconColor = 'ItemIconColor',
    ParentKey = 'ItemParentKey',
    IconOnly = 'ItemIconOnly',
    Visible = 'ItemVisible',
    Checked = 'ItemChecked',
    ItemHeader = 'ItemHeader',
    ItemTopDivider = 'ItemTopDivider',
    ItemDivider = 'ItemDivider',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
}

export const TextAlignmentTypes = {
    '0': 'center',
    '1': 'left',
    '2': 'right',
};
