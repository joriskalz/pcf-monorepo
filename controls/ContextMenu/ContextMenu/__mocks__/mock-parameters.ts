/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
