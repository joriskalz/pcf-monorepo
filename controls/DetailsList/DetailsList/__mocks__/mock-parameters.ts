/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
