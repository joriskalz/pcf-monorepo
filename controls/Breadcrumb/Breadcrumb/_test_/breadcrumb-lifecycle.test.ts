import { Breadcrumb } from '..';
import { IInputs } from '../generated/ManifestTypes';
import { ItemColumns } from '../ManifestConstants';
import { MockContext, MockState } from '../__mocks__/mock-context';
import { MockDataSet, MockEntityRecord } from '../__mocks__/mock-datasets';
import { getMockParameters } from '../__mocks__/mock-parameters';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';

// Since requestAnimationFrame does not exist in the test DOM, mock it
window.requestAnimationFrame = jest.fn().mockImplementation((callback) => {
    callback();
});

jest.useFakeTimers();

describe('BreadCrumb', () => {
    beforeEach(() => jest.clearAllMocks());
    afterEach(() => {
        for (let i = 0; i < document.body.children.length; i++) {
            if (document.body.children[i].tagName === 'DIV') {
                document.body.removeChild(document.body.children[i]);
                i--;
            }
        }
    });

    it('renders', () => {
        const { component, context } = createComponent();
        component.init(context);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('renders dummy items when no items configured', () => {
        const { component, context } = createComponent();
        // Simulate there being no items bound - which causes an error on the parameters
        context.parameters.items.error = true;
        component.init(context);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('raises the onSelect event', () => {
        const { component, context } = createComponent();

        component.init(context);
        const breadcrumbElement = component.updateView(context);

        const firstCommandReference = {
            id: { guid: '1' },
            name: '1',
        } as ComponentFramework.EntityReference;

        context.parameters.items.records['1'].getNamedReference = jest.fn().mockReturnValueOnce(firstCommandReference);
        const breadCrumb = mount(breadcrumbElement);
        const breadcrumbNode = breadCrumb.find('.ms-Breadcrumb-itemLink').first();
        expect(breadcrumbNode.length).toEqual(1);

        breadcrumbNode.simulate('click');
        expect(context.parameters.items.openDatasetItem).toBeCalledTimes(1);
        expect(context.parameters.items.openDatasetItem).toBeCalledWith(firstCommandReference);
    });

    it('theme', async () => {
        const { component, context, container } = createComponent();
        context.parameters.Theme.raw = JSON.stringify({
            palette: {
                themePrimary: '#0078d4',
            },
        });
        act(() => {
            component.init(context);
            component.updateView(context);
        });

        expect(container).toMatchSnapshot();
    });
});

function createComponent() {
    const component = new Breadcrumb();
    const notifyOutputChanged = jest.fn();
    const context = new MockContext<IInputs>(getMockParameters());
    context.parameters.items = new MockDataSet([
        new MockEntityRecord('1', {
            [ItemColumns.DisplayName]: 'Item 1',
            [ItemColumns.Key]: 'item1',
            [ItemColumns.Clickable]: true,
        }),
        new MockEntityRecord('2', {
            [ItemColumns.DisplayName]: 'Item 2',
            [ItemColumns.Key]: 'item2',
        }),
        new MockEntityRecord('3', {
            [ItemColumns.Key]: 'item7',
            [ItemColumns.DisplayName]: 'Item 7',
            [ItemColumns.Clickable]: true,
        }),
        new MockEntityRecord('4', {
            [ItemColumns.Key]: 'item8',
            [ItemColumns.DisplayName]: 'Item 8',
            [ItemColumns.Clickable]: true,
        }),
    ]);
    const state = new MockState();
    const container = document.createElement('div');
    document.body.appendChild(container);
    return { component, context, container, notifyOutputChanged, state };
}
