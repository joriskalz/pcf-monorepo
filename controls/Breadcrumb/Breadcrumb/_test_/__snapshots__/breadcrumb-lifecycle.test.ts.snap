// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb renders 1`] = `
<Memo(CanvasBreadCrumb)
  ariaLabel=""
  height={-1}
  items={
    Array [
      Object {
        "clickable": true,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "1",
          "values": Object {
            "ItemClickable": true,
            "ItemDisplayName": "Item 1",
            "ItemKey": "item1",
          },
        },
        "id": "1",
        "key": "item1",
        "text": "Item 1",
      },
      Object {
        "clickable": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "2",
          "values": Object {
            "ItemDisplayName": "Item 2",
            "ItemKey": "item2",
          },
        },
        "id": "2",
        "key": "item2",
        "text": "Item 2",
      },
      Object {
        "clickable": true,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "3",
          "values": Object {
            "ItemClickable": true,
            "ItemDisplayName": "Item 7",
            "ItemKey": "item7",
          },
        },
        "id": "3",
        "key": "item7",
        "text": "Item 7",
      },
      Object {
        "clickable": true,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "4",
          "values": Object {
            "ItemClickable": true,
            "ItemDisplayName": "Item 8",
            "ItemKey": "item8",
          },
        },
        "id": "4",
        "key": "item8",
        "text": "Item 8",
      },
    ]
  }
  maxDisplayedItems={4}
  onSelected={[Function]}
  overflowIndex={0}
  setFocus=""
  tabIndex={0}
  themeJSON=""
  width={-1}
/>
`;

exports[`BreadCrumb renders dummy items when no items configured 1`] = `
<Memo(CanvasBreadCrumb)
  ariaLabel=""
  height={-1}
  items={
    Array [
      Object {
        "clickable": true,
        "id": "1",
        "key": "1",
        "text": "Item 1",
      },
      Object {
        "clickable": true,
        "id": "2",
        "key": "2",
        "text": "Item 2",
      },
      Object {
        "clickable": true,
        "id": "3",
        "key": "3",
        "text": "Item 3",
      },
    ]
  }
  maxDisplayedItems={3}
  onSelected={[Function]}
  overflowIndex={0}
  setFocus=""
  tabIndex={0}
  themeJSON=""
  width={-1}
/>
`;

exports[`BreadCrumb theme 1`] = `<div />`;
